whylog/config/rule.py

```diff
@@ -19,10 +19,10 @@ def serialize(self):
             "constraints": self._constraints,
         }
 
-    def get_new_parsers(self, old_parsers):
+    def get_new_parsers(self, parser_name_generator):
         new_parsers = []
         for parser in itertools.chain([self._effect], self._causes):
-            if parser.name not in old_parsers:
+            if parser_name_generator.is_free_parser_name(parser.name):
```

Comment:
It is not very hard to fail writing with an exception and then front will have to handle it. There can be other write errors like no permissions, so Front needs to handle write errors anyway. Just raise an exception if the parser which you want to save is different from the one present in config already, if they have the same name.